go/common/ticker/ticker.go

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) 2024 Fantom Foundation
+//
+// Use of this software is governed by the Business Source License included
+// in the LICENSE file and at fantom.foundation/bsl11.
+//
+// Change Date: 2028-4-16
+//
+// On the date above, in accordance with the Business Source License, use of
+// this software will be governed by the GNU Lesser General Public License v3.
+
+package ticker
+
+import "time"
+
+//go:generate mockgen -source ticker.go -destination ticker_mocks.go -package ticker
+
+// Ticker is an abstraction of a ticker from standard time package.
+// It contains a channel which produces ticks at certain intervals
+// defined by implementations. For example, TimeTicker wraps
+// the standard time.Ticker based on time.Duration intervals.
+// Furthermore, the ticker includes the method Stop to terminate
+// the ticker. When the ticker is stopped, no more ticks will be
+// sent via the channel.
+type Ticker interface {
```

Comment:
nit: you should add to the documentation that all tickers need to be stopped -- only since `time.Ticker` Go 1.23 this is no longer needed; but I think we might not have set this yet as our minimum Go requirement.